node/runtime/src/lib.rs

```diff
@@ -1355,39 +1378,76 @@ mod tests {
         );
         assert_ne!(root, apply_result.root);
         runtime.state_db.commit(&mut apply_result.transaction).unwrap();
-        let result1 = viewer.view_account_at(hash(b"alice"), apply_result.root);
+        let result1 = viewer.view_account_at(&alice_account(), apply_result.root);
         assert_eq!(
             result1.unwrap(),
             AccountViewCallResult {
                 nonce: 1,
-                account: hash(b"alice"),
+                account: alice_account(),
                 amount: 90,
                 stake: 50,
                 code_hash: default_code_hash(),
             }
         );
-        let result2 = viewer.view_account_at(hash(b"eve"), apply_result.root);
+        let result2 = viewer.view_account_at(&eve_account(), apply_result.root);
         assert_eq!(
             result2.unwrap(),
             AccountViewCallResult {
                 nonce: 0,
-                account: hash(b"eve"),
+                account: eve_account(),
                 amount: 10,
                 stake: 0,
                 code_hash: hash(b""),
             }
         );
     }
 
+    #[test]
+    fn test_create_account_failure_invalid_name() {
+        let (mut runtime, viewer) = get_runtime_and_state_db_viewer();
+        let root = viewer.get_root();
+        let (pub_key, _) = get_keypair();
+        let tx_body = TransactionBody::CreateAccount(CreateAccountTransaction {
+            nonce: 1,
+            originator: alice_account(),
+            new_account_id: "eve".to_string(),
```

Comment:
It would make sense to test a most of incorrect accounts using a for loop, e.g. short, long, incorrect symbols, empty, non-ascii, with 0 inside.